fhirstore/fhirstore.py

```diff
@@ -290,75 +289,84 @@ def search(
-                    )
-                    # search the db for the specific resource to include
-                    included_hits = self.es.search(
-                        body={"simple_query_string": {"query": included_id, "fields": "id",}},
-                        index=f"fhirstore.{included_resource.lower()}",
-                    )
-            [
-                bundle["items"].append({"resource": h["_source"], "search": {"mode": "include"}})
-                for h in included_hits["hits"]["hits"]
-            ]
-
-        return bundle
-
-    def count(self, resource_type, params):
-        """Counts how many results match this query
-        """
-        self.validate_resource_type(resource_type)
+        # create a Parser instance and process the arguments
+        parsed_args = URL_Parser(args, resource_type)
+        parsed_args.process_params()
+
+        # Create an instance of querybuilder and build core query
+        core_query = CoreQueryBuilder(parsed_args.core_args)
+        core_query.build_core_query()
+
+        # Create an instance for the formatter and initial bundle
+        formatter = Formatter()
+        # TODO : find a way to avoid double parsing
+        formatter.initiate_bundle(args, resource_type)
+
+        # either do a count
+        if parsed_args.is_summary_count == True:
+            # Add new count to total results
+            hits = self.es.count(
+                body={"query": core_query.query},
+                index=f"fhirstore.{parsed_args.resource_type.lower()}",
+            )
+            formatter.fill_bundle(hits)
+        # or do a search
+        else:
+            query = {
+                "min_score": 0.01,
+                "from": parsed_args.offset,
+                "size": parsed_args.result_size,
+                "query": core_query.query,
+            }
+
+            if parsed_args.sort:
+                query["sort"] = parsed_args.sort
+
+            if parsed_args.elements:
+                query["_source"] = parsed_args.elements
+            print(query)
+            # .lower() is used to fix the fact that monstache changes resourceTypes to
+            # all lower case
+            hits = self.es.search(
+                body=query, index=f"fhirstore.{parsed_args.resource_type.lower()}"
+            )
 
-        core_query = build_core_query(params)
-        query = {"query": core_query}
-
-        # .lower() is used to fix the fact that monstache changes resourceTypes to
-        # all lower case
-        hits = self.es.count(body=query, index=f"fhirstore.{resource_type.lower()}")
-        return {
-            "resource_type": "Bundle",
-            "tag": {"code": "SUBSETTED"},
-            "total": hits["count"],
-        }
+            # add results to the bundle if they exist
+            formatter.fill_bundle(hits)
+        # TODO: Add case error when is_summary_count and include
+        if parsed_args.include and parsed_args.is_summary_count == False:
```

Comment:
I think this search function could be cut into pieces, this would at least improve readability and make unit testing easier